### PR TITLE
Clean Up Build

### PR DESCRIPTION
**Pull request by hazen**
- Removed the Scons files
- Updated build instructions for multiple platforms
- Build PBC .h and .c files from riak_pb instead of static versions
- Updated build version to 1.0
- Addresses https://github.com/basho/riak-c-client/issues/34, https://github.com/basho/riak-c-client/issues/35 and https://github.com/basho/riak-c-client/issues/36

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,75 +19,225 @@ significant changes to the API.**
...
-		or 
-	make PREFIX=/usr/local/
-	sudo make PREFIX=/usr/local/ install
-	
-	brew install protobuf protobuf-c libevent cunit
-	git clone git@github.com:/basho/riak-c-client.git
-	cd riak-c-client
-	./autogen.sh
-	make
+Be sure Xcode, command-line tools and [Home Brew](http://brew.sh/) installed first
```
Comment (bookshelfdave) [inline README.md]: add a period at the end

File: README.md [hunk 2]
```diff
@@ -22,75 +19,225 @@ significant changes to the API.**
 
 # Dependencies
 
-* autotools
+* automake
+* autoconf
+* libtool
 * libevent
 * protobuf
```
Comment (bookshelfdave) [inline README.md]: is it worth pointing out v2.5.0 here?

File: README.md [hunk 3]
```diff
@@ -22,75 +19,225 @@ significant changes to the API.**
...
 * protobuf-c
```
Comment (bookshelfdave) [inline README.md]: same with protobuf-c-0.15

**Comment by bookshelfdave** [comment]
does this PR bring RCC to v1.0? I see it stamped in `Makefile.am`.
If so, we'll need to tag/release etc.

File: README.md (hunk 2 above)
Comment (hazen) [inline README.md]: Yes

File: README.md (hunk 3 above)
Comment (hazen) [inline README.md]: Yes.   Same with libevent-2.0.21

**Comment by hazen** [comment]
Upped version to 1.0.  Agree we should tag when it's "done."

**Comment by bookshelfdave** [comment]
This builds fine if I follow the readme :-) 

I think we need to:
a) give the user a better error message if they haven't cloned riak_pb into this src tree
b) add a script + metadata file including riak_pb vsn# to automate the download
   (similar to rebar deps/locked deps)

**Comment by bookshelfdave** [comment]
I'd ask clients@basho about the version number. We probably want to avoid 1.0 in the configs until this is actually released.

**Comment by bookshelfdave** [comment]
@javajolt I pushed some readme updates directly to your branch. Can you give them a quick review?

**Comment by bookshelfdave** [comment]
:+1:

**Comment by hazen** [comment]
@metadave I don't see any changes to the README file on the `bugfix/bch/cleanup-build` branch. Did you push?

**Comment by bookshelfdave** [comment]
derp, pushed.

**Pull request merged**